xla/service/gpu/fusions/triton/triton_fusion_emitter_device_legacy_test.cc

```diff
@@ -2077,6 +2077,29 @@ ENTRY e {
   EXPECT_TRUE(RunAndCompare(kHloText, ErrorSpec{/*aabs=*/2e-3, /*arel=*/2e-3}));
 }
 
+TEST_F(TritonGemmLevel2Test, BroadcastOfScalarIsFused) {
```

Comment:
I don't think `IsFused` is what we want to test here. Rather, the test is about whether the emitter works correctly---i.e. given a post-optimization Triton fusion that contains the broadcast of an effective scalar, it lowers correctly to LLVM.

I think we can rewrite this test to only create IR and compile it down to LLVM. If you think it is necessary to check the numerics (I don't think it is), then `RunAndCompareNoHloPasses` can additionally be used.

Reply:
Sounds good - revised to check IR instead

Reply:
Thanks, this is a bit better---but note that `CreateTritonIrAndFileCheckForDot` does not actually compile all the way down to LLVM.

You will need to call `TritonWrapper` directly to do that, as is done in [triton_support_test.cc](https://github.com/openxla/xla/blob/e46379617054479acbb2453e3871b5b0d7b9c436/xla/service/gpu/fusions/triton/triton_support_test.cc#L191) (in the absence of a better testing API).

I think calling both `CreateTritonIrAndFileCheckForDot` as you did and `TritonWrapper` is a reasonable solution for now. I'll plan to make another change allowing to both check Triton IR & compile down to LLVM in a single call.

Reply:
Hi @elfiegg, any update on this?

Reply:
Sorry for the delay! Done testing in TritonWrapper too.

Reply:
gentle ping :)

Reply:
Hi @bchetioui - friendly ping ;:^)